### PR TITLE
fix included examples and markdown(book)

### LINKED ISSUE #911: Docs: Examples are broken and not showing
Screenshot 2024-03-27 at 11 06 36

On this page and a few others: https://rtic.rs/2/book/en/by-example/resources.html

### PR DESCRIPTION
**Pull request by enaut**
fixes: #911

### CODE REVIEW

File: book/en/src/by-example/app_init.md [hunk 1]
```diff
@@ -3,30 +3,31 @@
 An RTIC application requires an `init` task setting up the system. The corresponding `init` function must have the
 signature `fn(init::Context) -> (Shared, Local)`, where `Shared` and `Local` are resource structures defined by the user.
```
Comment (enaut) [inline book/en/src/by-example/app_init.md:4]: I'm just learning and not sure if true but I have a running example with a signature of:

```rust
    #[init]
    fn init(mut ctx: init::Context) -> (Shared, Local, init::Monotonics)
```
Is this paragraph outdated or incomplete?

File: book/en/src/by-example/tips/indirection.md [hunk 2]
```diff
@@ -13,14 +13,14 @@ As this example of approach goes completely outside of RTIC resource model with
 
 Here's an example where `heapless::Pool` is used to "box" buffers of 128 bytes.
 
-``` rust,noplayground
-{{#include ../../../../../rtic/examples/pool.rs}}
+```rust,noplayground
+{{#include ../../../../../examples/lm3s6965/examples/pool.rs_old}}
```
Comment (enaut) [inline book/en/src/by-example/tips/indirection.md]: This looks like that chapter is outdated?

File: book/en/src/by-example/app_init.md (hunk 1 above)
Comment (enaut) [reply book/en/src/by-example/app_init.md:4]: I see my example uses rtic v1 vs. rtic v2 here...

File: book/en/src/by-example/tips/indirection.md (hunk 2 above)
Comment (AfoHT) [reply book/en/src/by-example/tips/indirection.md]: @romancardenas Do you recall why `pool.rs` example got renamed to `pool.rs_old` as part of [PR815](https://github.com/rtic-rs/rtic/pull/815/files#diff-bfd7d183b2d78e4f0cb8d28f140120aa5428525ef9eba312cd6d89f2d83077e9)?

Comment (romancardenas) [reply book/en/src/by-example/tips/indirection.md]: No, my guess is that I did a refactor or something while moving the CI outside the rtic crate and I accidentally renamed the file.

Comment (AfoHT) [reply book/en/src/by-example/tips/indirection.md]: Then let's merge this, together with a renaming of the file. Skipping this last commit https://github.com/rtic-rs/rtic/pull/912/commits/d3692ca8b50d239d4cf8421a3ae73b828a49815b

**Review by AfoHT — APPROVED** [review]
@enaut Thank you for your great work! :)

I've handled the `pool` example, and while being at it done some other much needed docs cleanups

**Pull request merged**